### DISCUSSION TITLE
Webtransport for wasm

### DISCUSSION THREAD

**Discussion by cBournhonesque** [libp2p/rust-libp2p#5080] [Q&A]
Hi I saw that the webtransport transport was available for wasm.
However it looks like `datagrams` are not supported currently right?

**Comment by oblique** [comment]
Yes datagrams are not supported and WebTransport for libp2p doesn't have any usecase for them from the point of view of `Swarm`. Javascript and Go implementations doesn't use them too. In general libp2p is stream based.

Edit: Typo

**Reply by cBournhonesque** [reply]
Are there streams in webtransport for unreliable/unoredered data similar to datagrams? I wasn't able to get a clear answer from googling

**Reply by thomaseizinger** [reply]
In WebTransport yes, in libp2p no.

**Reply by thomaseizinger** [reply]
https://developer.mozilla.org/en-US/docs/Web/API/WebTransport_API#unreliable_transport
